sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/ResourceManagerTestBase.java

```diff
@@ -273,6 +276,49 @@ private void addTextReplacementRules(Map<String, String> rules) {
         }
     }
 
+    /**
+     * Sets sdk context when running the tests
+     *
+     * @param sdkContext the sdkContext
+     * @param objects the manager classes to change sdkContext
+     * @param <SdkContext> the type of sdk context
+     * @throws RuntimeException when field cannot be found or set.
+     */
+    protected <SdkContext> void setSdkContext(SdkContext sdkContext, Object... objects) {
```

Comment:
so no change to SdkContext itself right? I believe we all agree that SdkContext is confusing, need a better name, maybe test specific.

Reply:
I think the confusing part was because it was existing in `authenticate(httpPipeline, profile, sdkContext)`. It made users confused why it's required when creating the managers. After this change, now SdkContext is not settable for users. And we make it more difficult to find by moving from Manager(the base) to only ResourceManager/AuthorizationManager.

Reply:
BTW, why `<SdkContext>` as generic type?

Reply:
@weidongxu-microsoft We are adding this method in `azure-resourcemanager-test`. While `SdkContext` is existing in `azure-resourcemanager-resources`. Hence I have to use generic type here.